app/src/main/java/sdumchykov/androidApp/presentation/viewPager/myProfile/MyProfileFragment.kt

```diff
@@ -100,29 +110,29 @@ class MyProfileFragment :
 
     private fun imageViewMainProfilePictureSetOnClickListener() {
         binding.imageViewMainProfilePicture.setOnClickListener {
-//            activity?.viewModelStore.clear()
-
             val fetchContactList = !viewModel.getFetchContactList()
             viewModel.setFetchContactList(fetchContactList)
 
-            val toastText = if (fetchContactList) SHOW_CONTACT_LIST
+            if (fetchContactList) {
+                if (!isPermissionsGranted()) {
+                    FetchContacts().fetchContacts(
+                        activity as AppCompatActivity,
+                        { parentViewModel.initRealUsersList() },
+                        { parentViewModel.initHardcodedDataList() }
+                    )
+                } else parentViewModel.initRealUsersList()
+
+                if (!isPermissionsGranted()) {
+                    viewModel.setFetchContactList(false)
+                }
```

Comment:
You definitely should move this code to the other if check with the same condition. I can't see a reason to separate this snippet.